### ISSUE TITLE
SERVO42C Drivers will damage themselves when is they accidentally hit an endstop, current limit is not respected

### ISSUE THREAD

**Issue by foxalabs** [makerbase-mks/MKS-SERVO42C#43]
Hi,

I've been chatting with someone at MKS sales and they say it's my own fault that I have damaged my driver by hitting an endstop.

Q8 Q10 and Q11 have lets the magic smoke out and vapourised themselves after one of the drivers went off on it's ownb after calibrating and the resolution suggested? BUY A NEW ONE! How about you fix the issue that caused this driver to destroy itself with over current?

CNC and Laser machines regularly crash into endstops from either accidental user error or mechanical/electronic failure.

Stepper drivers should not catastrophically fail when reaching and endstop, the current limiter should engage and keep the device from self destructing.

I am willing to work with you and even do some code investigation.

**Comment by ghost** [comment]
Did you enable protection mode? 

Also before blaming others you should've configured your machine to respect endstops. 
-> `#define ENDSTOPS_ALWAYS_ON_DEFAULT`

It is NOT common for CNCs to crash. That happens because of user error and in industrial CNC mills a bent spindle caused by a crash can result in 10s of thousands repair cost. I can assure you that if you come at DMG, Mazak or Haas with a claim like that, they also tell you to spend some cash if you want to get it running again.

**Comment by foxalabs** [comment]
Yes, Endstop protection was ON. The Stepper had a mind of it's own, It made a hi pitched squealing noise and raced up to the top of the Z axis and then the end stop limit switch and carried on attempting to go up until I hit the E-stop and killed power to the controller board, in that time it had demolished itself. 

If crashing a tool into your work is not a common occurrence (or end stops) then why are there so many videos of it on youtube by all of the top manufactures and CNC youtubers?

If I damage a spindle by user error then I expect to pay for it, if the machine goes shooting off on its own then I expect the faulty equipment manufacturer to pay for the damage.
